src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,20 +179,33 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
-    boolean truncatedQuotedField = false;
+    boolean nextRecordTruncated = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
-      if (truncatedQuotedField) {
+      if (nextRecordTruncated) {
         onLastRow = true;
       }
 
-      if (!onLastRow) {
-        record = iterator.next();
+      try {
+        if (!onLastRow) {
+          record = iterator.next();
+        }
+      } catch (Exception e) {
+        /*
+         * FIX ME: If we hit an exception for a malformed row (in case of quotes for example), we mark this as the last
+         * row
+         */
+        if (e.getMessage() != null
+            && e.getMessage().contains("EOF reached before encapsulated token finished")) {
```

Comment:
Now that we're repeating this check, we should either put the exception message in a constant to avoid typos, or encapsulate the check in a helper.